src/averageAges.js

```diff
@@ -37,7 +43,15 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(woman => woman.sex === 'f');
+  const isMother = withChildren !== undefined
+    ? women.filter(woman => people.some(person => person.mother === woman.name))
```

Comment:
Move to the separate func & reuse it in all cases
```suggestion
    ? women.filter(woman => hasChild(people, woman.name))
```